src/test/java/com/google/sps/CapstoneAuthTest.java

```diff
@@ -50,54 +50,85 @@ public void setUp() throws NoSuchMethodException, InstantiationException, Illega
         mockUser = mock(UserRecord.class);
         PowerMockito.mockStatic(UserRecord.class);
 
-        mockToken = mock(FirebaseToken.class);
-        PowerMockito.mockStatic(FirebaseToken.class);
-
         when(FirebaseAuth.getInstance()).thenReturn(mockFirebase);
 
         when(mockFirebase.getUser(TEST_UID)).thenReturn(mockUser);
-        when(mockUser.getEmail()).thenReturn(TEST_EMAIL);
 
+        mockToken = mock(FirebaseToken.class);
+        PowerMockito.mockStatic(FirebaseToken.class);
+     
         when(mockFirebase.verifyIdToken(TEST_UID)).thenReturn(mockToken);
         when(mockToken.getUid()).thenReturn(TEST_ID_TOKEN);
 
-        obj = (CapstoneAuth) CapstoneAuth.getInstance();
+        capstoneAuth = (CapstoneAuth) CapstoneAuth.getInstance();
 
     }
 
     @Test
     public void testGetInstance_returnsValidInstance() throws IOException {
-        assertEquals(obj.getInstance(), obj);
+        assertSame(capstoneAuth.getInstance(), capstoneAuth);
     }
 
     @Test
-    public void testGetUserId_returnsValidId() throws IOException {
-        assertEquals(obj.getUserEmail(TEST_UID), TEST_EMAIL);
+    public void testGetUserEmail_returnsValidEmail() throws IOException {
+        when(mockUser.getEmail()).thenReturn(TEST_EMAIL);
+
+        assertEquals(capstoneAuth.getUserEmail(TEST_UID), TEST_EMAIL);
     }
 
 
     @Test(expected = NullPointerException.class)
-    public void testGetUserId_throwsNullPointerForFalseID() throws IOException {
-        obj.getUserEmail(TEST_FALSE_UID);
+    public void testGetUserEmail_throwsNullPointerForFalseID() throws IOException {
+        when(mockUser.getEmail()).thenReturn(TEST_EMAIL);
+
+        capstoneAuth.getUserEmail(TEST_FALSE_UID);
     } 
 
     @Test
-    public void testIsUserAuth_returnsTrue() throws IOException {
-       assertTrue(obj.isUserAuthenticated(TEST_UID));
+    public void testIsUserAuth_returnsTrue() throws IOException, FirebaseAuthException {
+
+       mockToken = mock(FirebaseToken.class);
+       PowerMockito.mockStatic(FirebaseToken.class);
+    
```

Comment:
nit: In unit tests, only add empty lines between the arrange/act/assert steps. Please remove the empty lines 89 and 92. Do the same for other tests

Reply:
Will do